### PR TITLE
Flatten sections without straplines into sections with straplines

### LINKED ISSUE #7842: Flatten FIX Page Sections
**Is your feature request related to a problem? Please describe.**
Add support for different section types
Refer to JIRA-1063 for further information

**Describe the solution you'd like**
If the strapline is missing, then flatten this section and add to end of the previous section
e.g. If the strapline is missing and the previous section is top stories, then flatten this section and add to top stories

#### Feature Index Page with no sections
https://app.zeplin.io/project/5c8a5e021d454124ef90669f/screen/5ee8cfb1ddde9f468b2998a0

#### Limit Max Number of Promos per section
See https://www.test.bbc.com/urdu/science-51314202?renderer_env=live
e.g. Top Stories = 13, Everything else = 10
For FIX Pages, the limits should be removed

#### TODO
- [ ] Determine how YouTube / external promo links should be rendered within a section e.g. https://www.test.bbc.com/urdu/science-51314202?renderer_env=live 
- [ ] Remove limits on max number of items to display on FIX pages only. Each section will only display 2 rows of 4 with images, and 1 row of 4 without images therefore the max can only be 12-14
- [ ] Refactor renderWithoutStrapline logic added in original PR https://github.com/bbc/simorgh/blob/706edd65525153496b768b2f0d47e197f0de4fc9/src/app/containers/IndexPageSection/index.jsx#L204
- [ ] Add more test scenarios for `flattenGroups` logic
- [ ] FIX page with no straplines is only displaying 13 Top Stories, and no additional sections - lots of content missing
- [ ] If a Top Stories section does not have 1, 5, 9 or 13, then some stories will be omitted

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### LINKED ISSUE #7892: h1 on Feature Index Page - should it have a span with lang
Related to #7714
Unit Tests may need to be updated accordingly

_Originally posted by @karinathomasbbc in https://github.com/bbc/simorgh/pull/7856#discussion_r492162260_

### PR DESCRIPTION
**Pull request by chewu**
Resolves #7842, #7892 

**Overall change:**
If the strapline is missing, then flatten this section and add to end of the previous section
e.g. If the strapline is missing and the previous section is top stories, then flatten this section and add to top stories

**Code changes:**
- Implement flattening of groups
- Adding tests & updating snapshots

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,7 +5,17 @@ SIMORGH_PUBLIC_STATIC_ASSETS_PATH=/
 SIMORGH_ATI_BASE_URL=https://logws1363.ati-host.net?
 SIMORGH_CONFIG_URL=https://config.test.api.bbci.co.uk/
 SIMORGH_INCLUDES_BASE_URL=https://www.test.bbc.com/ws/includes
+SIMORGH_INCLUDES_BASE_AMP_URL=https://news.test.files.bbci.co.uk
```
Comment (karinathomasbbc) [inline .env]: Eek, this might have been an accidental commit.. We'll need to revert this.

Comment (chewu) [reply .env]: oops, reverted.

File: src/app/containers/IndexPageSection/index.jsx [hunk 2]
```diff
@@ -95,8 +95,8 @@ const parentGridColumns = {
   group5: 8,
 };
 
-const renderPromos = (items, isFirstSection, dir) => {
-  const rows = getRows(items, isFirstSection);
+const renderPromos = ({ items, isFirstSection, dir, showAllRows }) => {
```
Comment (karinathomasbbc) [inline src/app/containers/IndexPageSection/index.jsx]: Could we rename this prop to `showAllPromos`?

Comment (chewu) [reply src/app/containers/IndexPageSection/index.jsx]: Done.

File: src/app/containers/IndexPageSection/utilities/storyRowsSplitter.jsx [hunk 3]
```diff
@@ -23,11 +23,17 @@ const splitNoImageRow = standardSlices => [
...
-  );
+  let regularRows = [];
+  let noImageRow = [];
+  if (showAllPromos) {
+    regularRows = splitStandardRows(unsplitregularItems);
+  } else {
+    [regularRows, [noImageRow]] = splitNoImageRow(
+      splitStandardRows(unsplitregularItems),
+    );
+  }
```
Comment (karinathomasbbc) [inline src/app/containers/IndexPageSection/utilities/storyRowsSplitter.jsx]: I'd like to refactor this a little bit to make it a bit easier to read / maintain.

```suggestion
  let regularRows = [];
  let noImageRow = [];
  const standardRows = splitStandardRows(unsplitregularItems);
  
  if (showAllPromos) {
    regularRows = standardRows;
  } else {
    [regularRows, [noImageRow]] = splitNoImageRow(standardRows);
  }
```

File: src/app/containers/IndexPageSection/utilities/storyRowsSplitter.test.jsx [hunk 4]
```diff
@@ -259,13 +259,17 @@ const numberOfStories = {
```
Comment (karinathomasbbc) [inline src/app/containers/IndexPageSection/utilities/storyRowsSplitter.test.jsx:474]: We should add a couple of tests here to test the behaviour when `showAllPromos` is true

**Comment by karinathomasbbc** [comment]
We also need to fix up unit tests & integration tests

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/containers/IndexPageSection/index.jsx [hunk 5]
```diff
@@ -197,14 +197,6 @@ const IndexPageSection = ({
   // This may change in the future, if a way to avoid breaking UX is found.
   // Also, don't render a section without any items.
 
-  // If renderWithoutStrapline is false (which it is by default),
-  // then the section will not render if there is no strapline.
-  // If renderWithoutStrapline is true, and there is no strapline present,
-  // the section will always render.
-  if (!strapline && !renderWithoutStrapline) {
```
Comment (karinathomasbbc) [inline src/app/containers/IndexPageSection/index.jsx:204]: I think we still need to keep the `!strapline` logic here to maintain backwards compatibility with Home & IDX pages

**Review by rhenshaw56 — APPROVED** [review]

File: src/app/containers/IndexPageSection/index.jsx (hunk 5 above)
Comment (chewu) [reply src/app/containers/IndexPageSection/index.jsx:204]: Forgot about that, putting back in.

**Pull request merged**